src/compound_proof.rs

```diff
@@ -0,0 +1,76 @@
+use crate::hash::Algorithm;
+use crate::proof::Proof;
+
+use std::marker::PhantomData;
+use typenum::marker_traits::Unsigned;
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct CompoundProof<T: Eq + Clone + AsRef<[u8]>, U: Unsigned, N: Unsigned> {
```

Comment:
please document how this proof type works, and what exactly the fields in it are